modules/affile/affile-dest.c

```diff
@@ -191,8 +203,12 @@ affile_dw_reopen(AFFileDestWriter *self)
                 evt_tag_str("filename", self->filename),
                 evt_tag_errno(EVT_TAG_OSERROR, errno),
                 NULL);
+      if(!self->reopen_pending)
+        affile_dw_arm_reopen(self);
+      g_static_mutex_unlock(&self->lock);
       return self->owner->super.super.optional;
     }
+  g_static_mutex_unlock(&self->lock);
   return TRUE;
 }
 
```

Comment:
Please note that we released self->lock in affile_dw_queue() in order to avoid stalling the LogWriter queue until the reopen happens, e.g. we are queueing messages even during the reopen. This may or may not have a performance impact. (probably not so much).

This patch readds the lock around the entire reopen operation, however a couple of things to take care of:
* if we really enclose reopen into the lock, then we should probably drop the locking around the dw_reopen() call within dw_queue to avoid acquire/release three times on the same lock.
* we should also check whether the main_loop_call() within reopen would cause a deadlock or not. Locks held during main_loop_call() are always suspicious.
* this creates as many timers as there are file destinations, thus thousands of them. What is the gain here? Isn't it enough to do this kind of polling within the reap timer?